packages/esm-outpatient-app/src/transition-queue-entry/transition-queue-entry-dialog.component.tsx

```diff
@@ -0,0 +1,135 @@
+  const { t } = useTranslation();
+
+  const [selectedQueueLocation, setSelectedQueueLocation] = useState(queueEntry?.queueLocation);
+  const config = useConfig() as ConfigObject;
+  const defaultTransitionStatus = config.concepts.defaultTransitionStatus;
+
+  const preferredIdentifiers = queueEntry?.identifiers.filter((identifier) =>
+    config.defaultIdentifierTypes.includes(identifier?.identifierType?.uuid),
+  );
+
+  const startDate = dayjs(new Date().toISOString()).subtract(6, 'month').toISOString();
+  const { upcomingAppointment, isLoading } = usePatientAppointments(
+    queueEntry?.patientUuid,
+    startDate,
+    new AbortController(),
+  );
+  const { visits, isLoading: loading } = usePastVisits(queueEntry?.patientUuid);
+
+  const launchEditPriorityModal = useCallback(() => {
+    const endedAt = toDateObjectStrict(toOmrsIsoString(new Date()));
+    updateQueueEntry(
+      queueEntry?.visitUuid,
+      queueEntry?.queueUuid,
+      queueEntry?.queueUuid,
+      queueEntry?.queueEntryUuid,
+      queueEntry?.patientUuid,
+      queueEntry?.priorityUuid,
+      defaultTransitionStatus,
+      endedAt,
+      queueEntry?.sortWeight,
+      new AbortController(),
+    ).then(
+      ({ status }) => {
+        if (status === 201) {
+          showToast({
+            critical: true,
+            title: t('success', 'Success'),
+            kind: 'success',
+            description: t('patientAttendingService', 'Patient attending service'),
+          });
+          closeModal();
+          mutate(`/ws/rest/v1/visit-queue-entry?location=${selectedQueueLocation}&v=full`);
+          navigate({ to: `\${openmrsSpaBase}/patient/${queueEntry?.patientUuid}/chart` });
+        }
+      },
+      (error) => {
+        showNotification({
+          title: t('queueEntryUpdateFailed', 'Error updating queue entry'),
+          kind: 'error',
+          critical: true,
+          description: error?.message,
+        });
+      },
+    );
+  }, [queueEntry]);
+
+  return (
+    <div>
+      <ModalHeader closeModal={closeModal} title={t('admitPatient', 'Admit patient')} />
+      <ModalBody className={styles.modalBody}>
+        <div>
+          <section className={styles.modalBody}>
+            <p className={styles.p}>
+              {t('patientName', 'Patient name')} : &nbsp; {queueEntry?.name}
+            </p>
+            {preferredIdentifiers?.length
+              ? preferredIdentifiers.map((identifier) => (
+                  <p className={styles.p}>
+                    {identifier?.identifierType?.display} : &nbsp; {identifier?.identifier}
+                  </p>
+                ))
+              : ''}
+            <p className={styles.p}>
+              {t('lastClinicalVisit', 'Last clinical visit')} : &nbsp;{' '}
+              {loading
+                ? t('loading', 'Loading...')
+                : !visits
+                ? t('none', 'None')
+                : formatDatetime(parseDate(visits?.startDatetime))}
```

Comment:
using guard clause here would be better and easier to maintain check this [article](https://medium.com/nirman-tech-blog/using-guard-clauses-for-cleaner-react-code-77360e648659)